server/ai_http.go

```diff
@@ -86,36 +89,66 @@ func aiHttpHandle[I any](h *lphttp, decoderFunc func(*I, *http.Request) error) h
 			respondWithError(w, err.Error(), http.StatusBadRequest)
 			return
 		}
-
 		handleAIRequest(ctx, w, r, orch, req)
 	})
 }
 
 func (h *lphttp) StartLiveVideoToVideo() http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-
 		remoteAddr := getRemoteAddr(r)
 		ctx := clog.AddVal(r.Context(), clog.ClientIP, remoteAddr)
 
-		// skipping handleAIRequest for now until we have payments
-
-		var (
-			mid        = string(core.RandomManifestID())
-			pubUrl     = TrickleHTTPPath + mid
-			subUrl     = pubUrl + "-out"
-			controlUrl = pubUrl + "-control"
-		)
-		jsonData, err := json.Marshal(
-			&worker.LiveVideoToVideoResponse{
-				PublishUrl:   pubUrl,
-				SubscribeUrl: subUrl,
-				ControlUrl:   controlUrl,
-			})
-		if err != nil {
-			respondWithError(w, err.Error(), http.StatusInternalServerError)
+		var req worker.GenLiveVideoToVideoJSONRequestBody
+		if err := jsonDecoder[worker.GenLiveVideoToVideoJSONRequestBody](&req, r); err != nil {
+			respondWithError(w, err.Error(), http.StatusBadRequest)
 			return
 		}
 
+		// `startStream` is sent on the 2nd request from to the gateway to hand-off the running stream to the ai-runner
```

Comment:
I am still not sure why a second request is necessary; the O already has all the parameters  and the G is just sending those back. Why not  submit those params to the worker as part of processing the first request?

Reply:
If you do not want to modify the worker to return params back  to the G (in order to use `handleAIRequest`) then you can copy out the relevant parts from `handleAIRequest` and invoke the required calls directly; AFAICS you're left with only a capacity check and `orch.LiveVideoToVideo`

Most of `handleAIRequest` is boilerplate that can be omitted for now (specialized pipeline handling, payments, metrics)

Reply:
Makes sense, I will modify the flow to exclude `handleAIRequest` and try to do this in a single request. It first seemed there was a context issue with starting the trickle stream before the end of the http request which led to the handoff to fail, but I think `handleAIRequest` was just interfering with the response

Reply:
@j0sh I've updated `StartLiveVideoToVideo` to include the relevant parts from `handleAIRequest`